### DISCUSSION TITLE
[Feature Request] Add stack for overlay views

### DISCUSSION THREAD

**Discussion by KyuubiRan** [ppy/osu#29870] [Ideas] [answered]
Request to add stack for overlay views, current logic is so terrible!

**Expect:**
UserProfileOverlay `=Push stack=>` BeatmapOverlay `=Pop stack=>` UserProfileOverlay

**Actual:**
UserProfileOverlay `=Change to=>` BeatmapOverlay `=Close=>` Close Overlay

https://github.com/user-attachments/assets/b90a266d-4244-4fce-a2e8-e4335056b4e1

**Comment by Joehuu** [comment, marked as answer]
Already tracked as an issue: https://github.com/ppy/osu/issues/5063
